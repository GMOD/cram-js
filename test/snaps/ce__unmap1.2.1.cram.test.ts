import { testFile } from '../testUtil'
testFile('ce#unmap1.2.1.cram')
