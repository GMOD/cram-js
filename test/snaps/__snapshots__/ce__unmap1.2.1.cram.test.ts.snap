// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole ce#unmap1.2.1.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 2,
    "minorVersion": 1,
  },
  {
    "containerHeader": {
      "_endPosition": 39,
      "_size": 13,
      "alignmentSpan": 0,
      "landmarks": [
        0,
      ],
      "length": 10819,
      "numBases": 0,
      "numBlocks": 1,
      "numLandmarks": 1,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 39,
        "_size": 13,
        "alignmentSpan": 0,
        "landmarks": [
          0,
        ],
        "length": 10819,
        "numBases": 0,
        "numBlocks": 1,
        "numLandmarks": 1,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "VN",
                "value": "1.0",
              },
              {
                "tag": "SO",
                "value": "unsorted",
              },
            ],
            "tag": "HD",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "bowtie2",
              },
              {
                "tag": "PN",
                "value": "bowtie2",
              },
              {
                "tag": "VN",
                "value": "2.0.0-beta5",
              },
            ],
            "tag": "PG",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_II",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "8e7993f7a93158587ee897d7287948ec",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_III",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "3adcb065e1cf74fafdbba1e8c352b323",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_IV",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "251af66a69ee589c9f3757340ec2de6f",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_V",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "cf200a65fb754836dcc56b24b3170ee8",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_X",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "6f9368fd2192c89c613718399d2d31fc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_MtDNA",
              },
              {
                "tag": "LN",
                "value": "5000",
              },
              {
                "tag": "M5",
                "value": "cd05857ece6411f40257a565ccfe15bb",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "UNKNOWN",
              },
              {
                "tag": "SM",
                "value": "UNKNOWN",
              },
            ],
            "tag": "RG",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BA": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 30,
          },
          "parametersBytes": 1,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              4,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              100,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {
        "YTZ": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 9,
            "stopByte": 9,
          },
          "parametersBytes": 2,
        },
      },
      "tagIdsDictionary": [
        [
          "YTZ",
        ],
      ],
    },
    "containerHeader": {
      "_endPosition": 10877,
      "_size": 19,
      "alignmentSpan": 1,
      "landmarks": [
        178,
      ],
      "length": 823,
      "numBases": 1000,
      "numBlocks": 7,
      "numLandmarks": 1,
      "numRecords": 10,
      "recordCounter": 0,
      "refSeqId": -1,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 10877,
        "_size": 19,
        "alignmentSpan": 1,
        "landmarks": [
          178,
        ],
        "length": 823,
        "numBases": 1000,
        "numBlocks": 7,
        "numLandmarks": 1,
        "numRecords": 10,
        "recordCounter": 0,
        "refSeqId": -1,
        "refSeqStart": 0,
      },
      {
        "features": [
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              32,
              34,
              34,
              34,
              34,
              34,
              32,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              33,
              26,
              34,
              34,
              34,
              32,
              32,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              35,
              28,
              33,
              24,
              33,
              34,
              32,
              33,
              34,
              33,
              34,
              33,
              29,
              33,
              33,
              33,
              34,
              31,
              33,
              27,
              27,
              31,
              33,
              32,
              26,
              33,
              34,
              34,
              30,
              33,
              29,
              32,
              27,
              27,
              31,
              7,
              30,
              25,
              19,
              28,
              28,
              19,
            ],
            "readBases": "GATCGCACTGAAAATCTGGATATAGAACGTGTGCAAATGATTGTCTCTACCGTTCCGTAAAAATTATTGCTAATTAGCAATGATTTTAAGCTAATTAGTT",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808391",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11061,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              0,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
            ],
            "readBases": "NCATCTTCTGCAGGCGCGGTCGGCACCGTGTATTCGTTACGTTCCACGCCGATGATATTAGACACCAGCACGGTGTGATGCGCAGTCACCGCACGACCCG",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808433",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11062,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              0,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
            ],
            "readBases": "NGTTTGAGCGCAATGACTTGACGATGACCGGTGATTACAGTAACCAGCATATTGTGCCGATGAAGCAGGCTGTCGCTCCGCAATTTGAAGAGGGACACGA",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808462",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11063,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              0,
              8,
              8,
              8,
              8,
              8,
              8,
              6,
              6,
              5,
              20,
              23,
              20,
              20,
              20,
              20,
              32,
              32,
              32,
              32,
              32,
              21,
              32,
              32,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
            ],
            "readBases": "NGCTTCCCACTATATTTTACTCTCTGTGGCTTCCCACTTTATTTTACTCTCTGTGGCTTTATCGATTATTTTTTTCTCCTTGGCTTTCCTCTTTATTTTT",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808468",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11064,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              0,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
            ],
            "readBases": "NCGTCTTCTGCGGTTTCTATGATAGAGCAATTCTGTGGCGGATATTAAGATAACGGGTATCTGTTCTTTAATGATGCCGCAAAAAATGGAAAGTTAAGAA",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808471",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11065,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              0,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
            ],
            "readBases": "NGGAAATTCTGCGGTATCGAGAGATATTTGCCTTTAATTACCCCATAGCCGATTAACATCCCGGCACTATTAATCATTTCAATATTATCAATCACAAAGT",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808489",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11066,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              0,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
            ],
            "readBases": "NTGGAACACCATGCCCACTTTCGCACGCAGCAGGGCGATATCCTGAGAGTTGGTCAGGATGTTGTCGCCATCAAGCAGAATTTCACCTTCCGCACGCTGC",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808504",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11067,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              0,
              9,
              9,
              9,
              7,
              8,
              7,
              10,
              10,
              10,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              25,
              22,
              23,
              25,
              23,
              18,
              16,
              23,
              25,
              25,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
            ],
            "readBases": "NCAGAAATATATTGAACTTTTGAAACATGTCCCAACTTTTTTTTTTTTTAGAACACGTTGGCTGGTTTTTCAAGTTTTGAATACTTTTTAGGTTTTTTTG",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808511",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11068,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              0,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
            ],
            "readBases": "NTGTCAATACATTGTAAGGATTTCCCATTATGTCAATACATAGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATTTCGTATGTCGTGTTTCTTT",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808521",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11069,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              0,
              9,
              9,
              9,
              10,
              12,
              12,
              11,
              6,
              8,
              32,
              32,
              32,
              32,
              22,
              22,
              23,
              20,
              22,
              22,
              32,
              22,
              17,
              32,
              32,
              20,
              15,
              23,
              23,
              23,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              22,
              17,
              32,
              23,
              23,
              23,
              20,
              23,
              32,
              32,
              32,
              32,
              22,
              22,
              32,
              32,
              32,
              32,
              32,
              32,
              22,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              32,
              23,
              32,
              32,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
            ],
            "readBases": "NTACTTCTTACAAGATTTTTATCGCTACTTGAAAAAAATTCATCAACATAATCGCCATGACTATCTGCACCGTGAGTATGATAAGCAGCAACTCTATCTG",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "SRR065390.33808523",
            "sequenceId": -1,
            "tags": {
              "YT": "UU",
            },
            "templateSize": 0,
            "uniqueId": 11070,
          },
        ],
        "header": {
          "_endPosition": 11096,
          "_size": 36,
          "compressedSize": 36,
          "compressionMethod": "raw",
          "content": Uint8Array [
            255,
            255,
            255,
            255,
            15,
            0,
            1,
            10,
            0,
            5,
            4,
            9,
            11,
            12,
            30,
            255,
            255,
            255,
            255,
            15,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
          ],
          "contentId": 0,
          "contentPosition": 11060,
          "contentType": "MAPPED_SLICE_HEADER",
          "parsedContent": {
            "_endPosition": 10913,
            "_size": 36,
            "contentIds": [
              9,
              11,
              12,
              30,
            ],
            "md5": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "numBlocks": 5,
            "numContentIds": 4,
            "numRecords": 10,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": -1,
            "refSeqSpan": 1,
            "refSeqStart": 0,
          },
          "uncompressedSize": 36,
        },
      },
    ],
  },
]
`;
